diagrams/generate.dart

```diff
@@ -308,18 +336,121 @@ class DiagramGenerator {
     return files;
   }
 
+  Map<String, dynamic> _loadMetadata(File metadataFile) {
+    if (!metadataFile.isAbsolute) {
+      metadataFile = new File(
+        path.normalize(
+          path.join(temporaryDirectory.absolute.path, metadataFile.path),
+        ),
+      );
+    }
+    assert(metadataFile.existsSync());
+    final Map<String, dynamic> metadata = json.decode(metadataFile.readAsStringSync());
+    final String baseDir = path.dirname(metadataFile.absolute.path);
+    final List<File> frameFiles = metadata['frame_files']
+        .map<File>((dynamic name) => new File(path.normalize(path.join(baseDir, name)))).toList();
+    metadata['frame_files'] = frameFiles;
+    metadata['metadata_file'] = path.normalize(metadataFile.absolute.path);
+    return metadata;
+  }
+
+  Stream<List<int>> _concatInputs(List<File> files) async* {
+    for (File file in files) {
+      final Stream<List<int>> fileStream = file.openRead();
+      await for (List<int> block in fileStream) {
+        yield block;
+      }
+    }
+  }
+
+  Future<List<File>> _buildMoviesFromMetadata(List<Map<String, dynamic>> metadataList) async {
+    final Directory destDir = new Directory(path.joinAll(path.split(projectDir)..removeLast()));
+    final List<File> outputs = <File>[];
+    for (Map<String, dynamic> metadata in metadataList) {
+      final String prefix = '${metadata['category']}/${metadata['name']}';
+      final File destination = new File(path.join(destDir.path, '$prefix.mp4'));
+      if (destination.existsSync()) {
+        destination.deleteSync();
+      }
+      print('Converting ${metadata['name']} animation to mp4.');
+      await processRunner.runProcess(
+        <String>[
+          ffmpegCommand,
```

Comment:
can we document somewhere that you need ffmpeg? what happens if you run this without it?

Reply:
Sure. Added it to the README.md list of prerequisites.

If you run it without it, it will throw an exception and the script will fail.